schema-dex-agg.graphql

```diff
@@ -0,0 +1,514 @@
+  cumulativeSupplySideRevenueUSD: BigDecimal!
+
+  " Gross revenue for the protocol (revenue claimed by protocol). Examples: AMM protocol fee (Sushi’s 0.05%). OpenSea 10% sell fee. "
+  cumulativeProtocolSideRevenueUSD: BigDecimal!
+
+  " All revenue generated by the protocol. e.g. 0.30% of swap fee in Sushiswap, all yield generated by Yearn. "
+  cumulativeTotalRevenueUSD: BigDecimal!
+
+  " Number of cumulative unique users "
+  cumulativeUniqueUsers: Int!
+
+  " The fee type for this protocol "
+  feeType: FeeType!
+
+  ##### Snapshots #####
+
+  " Daily usage metrics for this protocol "
+  dailyUsageMetrics: [UsageMetricsDailySnapshot!]!
+    @derivedFrom(field: "protocol")
+
+  " Hourly usage metrics for this protocol "
+  hourlyUsageMetrics: [UsageMetricsHourlySnapshot!]!
+    @derivedFrom(field: "protocol")
+
+  " Daily financial metrics for this protocol "
+  financialMetrics: [FinancialsDailySnapshot!]! @derivedFrom(field: "protocol")
+}
+
+#########################
+##### Virtual Pools #####
+#########################
+"""
+Virtual Pools represent swapped token groups. DEX Aggs use liquidity pools from other protocols,
+so we we use virtual pools to track metrics for swaps betwee specific token groups.
+"""
+type VirtualPool @entity {
+  " { `InputTokens: {Sorted token addresses separated by `-`}` + `OutputTokens: {Sorted token addresses separated by `-`}` } "
+  id: ID!
+
+  " Timestamp of the last swap of this type "
+  timestamp: BigInt!
+
+  " Block number of the last swap of this type "
+  blockNumber: BigInt!
+
+  " Input Tokens "
+  tokensIn: [Token!]!
+
+  " Output Tokens "
+  tokensOut: [Token!]!
+
+  " Field to track the cumulative volume of the inward multi token in native amounts "
+  cumulativeInVolumes: [BigInt!]!
+
+  " Field to track the cumulative volume of the inward multi token in USD "
+  cumulativeInVolumesUSD: [BigDecimal!]!
+
+  " Field to track the cumulative volume of the outward multi token in native amounts "
+  cumulativeOutVolumes: [BigInt!]!
+
+  " Field to track the cumulative volume of the outward multi token in USD "
+  cumulativeOutVolumesUSD: [BigDecimal!]!
+
+  " Field to track cumulative volume of multi token summed across all tokens in USD "
+  cumulativeVolumeUSD: BigDecimal!
+
+  ### Revenues do not count revenues towards to underlying DEXs used to execute the swap. ###
+
+  " All revenue generated by the virtual pool, accrued to the supply side. "
+  cumulativeSupplySideRevenueUSD: BigDecimal!
+
+  " All revenue generated by the virtual pool, accrued to the protocol. "
+  cumulativeProtocolSideRevenueUSD: BigDecimal!
+
+  " All revenue generated by the virtual pool. "
+  cumulativeTotalRevenueUSD: BigDecimal!
+}
+
+###################################
+##### Virtual Pool Timeseries #####
+###################################
+
+type VirtualPoolDailySnapshot @entity {
```

Comment:
It was something that was important to the Odos team to have (I built an unofficial DEX Agg subgraph for them). I can take it out if there is consensus of it being overkill, but it seemed to me that since it was a request from people using the data that it might be worth keeping in as a standard.